### PR TITLE
Merge latest master into distributed tracing branch

### PR DESCRIPTION
**Pull request by ewertons**
# Checklist
- [x] I have read the [contribution guidelines] (https://github.com/Azure/azure-iot-sdk-c/blob/master/.github/CONTRIBUTING.md).
- [ ] I added or modified the existing tests to cover the change (we do not allow our test coverage to go down).
- If this is a modification that impacts the behavior of a public API
  - [ ] I edited the corresponding document in the `devdoc` folder and added or modified requirements.
- I submitted this PR against the correct branch: 
  - [ ] This pull-request is submitted against the `master` branch. 
  - [x] I have merged the latest `master` branch prior to submission and re-merged as needed after I took any feedback.
  - [ ] I have squashed my changes into one with a clear description of the change.

# Reference/Link to the issue solved with this PR (if any)

# Description of the problem

# Description of the solution

### CODE REVIEW

File: build_all/docs/Doxyfile [hunk 1]
```diff
@@ -38,7 +38,7 @@ PROJECT_NAME           = "Microsoft Azure IoT Device SDK for C"
 # could be handy for archiving the generated documentation or if some version
 # control system is used.
 
-PROJECT_NUMBER         = 1.3.9
+PROJECT_NUMBER         = 1.6.0
```
Comment (CIPop) [inline build_all/docs/Doxyfile:41]: Shouldn't this be something higher? We are already at 1.6.0 in `master`.

File: iothub_client/inc/internal/iothub_internal_consts.h [hunk 2]
```diff
@@ -9,11 +9,7 @@ extern "C"
 {
 #endif
 
-    static const char* IOTHUB_API_VERSION = "2019-10-01";
-    // TODO: https://github.com/Azure/azure-iot-sdk-c/issues/1547 tracks removing this preview 
-    // variable once the underlying logic is enabled on all IoTHubs.
-    static const char* IOTHUB_API_PREVIEW_VERSION = "2020-05-31-preview";
-
+    static const char* IOTHUB_API_VERSION = "2020-09-30";
```
Comment (CIPop) [inline iothub_client/inc/internal/iothub_internal_consts.h:12]: Does this have full-support for distributed tracing or do we need to use a different -preview APIversion?

**Review by CIPop — APPROVED** [review]
Left a couple of comments on versions.

File: build_all/docs/Doxyfile (hunk 1 above)
Comment (ewertons) [reply build_all/docs/Doxyfile:41]: For public-preview we usually keep the same version as in master. Maybe another point to fix on the versioning strategy?

File: iothub_client/inc/internal/iothub_internal_consts.h (hunk 2 above)
Comment (ewertons) [reply iothub_client/inc/internal/iothub_internal_consts.h:12]: That's something that needs to be verified. Notice that "IOTHUB_API_PREVIEW_VERSION" was introduced by John for pnp, so it is not related to distributed-tracing.

**Pull request merged**